### ISSUE TITLE
magit-process with section & output

### ISSUE THREAD

**Issue by Silex** [magit/magit#4586]
Hello,

I'm trying to use `magit-call-process` or `magit-start-process` AND get the output of the command. I see there's also `magit-process-file` but I don't think that could help me.

Basically it looks like `magit-process` is great to run a command and then refresh some buffers but when it comes to actually handling its output all commands seems to re-roll something custom, like `magit-parse-git-async` or `magit-git-string`.

Am I mistaken here?

That said, I could already use `magit-start-process` for all commands that only need to refresh the buffer once executed.

The case where I need output & async I guess I could do something like this for git tag:

``` elisp
  (magit-run-git-async "tag" args (and msg (list "-m" msg)) tag)
  (set-process-sentinel
   magit-this-process
   (lambda (process event)
     (when (memq (process-status process) '(exit signal))
       (magit-process-sentinel process event)
       (magit-refs-setup-buffer "HEAD" (magit-show-refs-arguments))))))
```

So, basically I'd call `magit-start-process` and use a custom filter & sentinel. Basically a filter that collects output & a sentinel that calls a custom callback with the output when the process is done. The custom filter would also call `magit-process-filter` so I have the output shown in the `$` buffer.

That said I realise that most of the time when I want output I don't _really_ want the output to show in the `$` buffer.

I'll play with it more, sorry for the rubber-ducking in this issue :wink:

If you have any hints they'd be appreciated otherwise just close this issue.

**Comment by Silex** [comment]
I found out that calling `magit-start-process` uses `magit-process-setup` which uses `magit-process-buffer` which uses `magit-toplevel` which makes it very complicated to use my own buffer.

Anyway it made me learn about processes, sentinels & filters and I think I'd be good to go rolling my own, I'll probably reuse things like `magit-process-insert-section` and friends.

**Comment by tarsius** [comment]
Good luck :grinning:
